### DISCUSSION TITLE
When fetching huge amounts of data, is there a large overhead?

### DISCUSSION THREAD

**Discussion by birgersp** [prisma/prisma#23914] [Q&A]
### Question

For some particular queries, I use $queryRawUnsafe to fetch data. For some of these queries, the size of the batch returned is very very large. I want to try to speed things up a bit. Indices are in place and working but I'm wondering if perhaps I could make it even faster.

In the current implementation, the data ends up as an array of objects (JSON). So I am wondering if perhaps for these particular queries I'd be better of using a library that doesn't convert data to JSON.

What's your take on this? Does Prisma have a "large" overhead when fetching data? Or would I not really notice the difference, with some library that would give me more "low-level" access to the fetched data?

**Comment by ludralph** [comment]
Hi @birgersp 👋 

Thank you raising this question.

> What's your take on this? Does Prisma have a "large" overhead when fetching data?

Prisma Client does introduce some overhead when fetching data, especially when dealing with large datasets. This overhead is due to the operations Prisma Client performs such as figuring out what the query means, generating the SQL for it, executing it, fetching the data, transforming the data into a format that JavaScript can understand, and then returning the data to the application.

To get a better understanding of what's causing the performance issue, you could try enabling [tracing](https://www.prisma.io/docs/orm/prisma-client/observability-and-logging/opentelemetry-tracing). Tracing provides a detailed log of the activity that Prisma Client carries out, including the time taken to execute each query.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
